airbyte-db/src/main/java/io/airbyte/db/bigquery/BigQueryUtils.java

```diff
@@ -0,0 +1,130 @@
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.google.cloud.bigquery.Field;
+import com.google.cloud.bigquery.FieldList;
+import com.google.cloud.bigquery.FieldValue;
+import com.google.cloud.bigquery.FieldValue.Attribute;
+import com.google.cloud.bigquery.FieldValueList;
+import com.google.cloud.bigquery.LegacySQLTypeName;
+import com.google.cloud.bigquery.QueryParameterValue;
+import com.google.cloud.bigquery.StandardSQLTypeName;
+import io.airbyte.commons.json.Jsons;
+import io.airbyte.db.DataTypeUtils;
+import io.airbyte.protocol.models.JsonSchemaPrimitive;
+import java.text.DateFormat;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Collections;
+import java.util.Date;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class BigQueryUtils {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(BigQueryUtils.class);
+
+  public static final DateFormat BIG_QUERY_DATE_FORMAT = new SimpleDateFormat("yyyy-MM-dd");
+  public static final DateFormat BIG_QUERY_DATETIME_FORMAT = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss");
+  public static final DateFormat BIG_QUERY_TIMESTAMP_FORMAT = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSSSS z");
+
+  public static JsonNode rowToJson(FieldValueList rowValues, FieldList fieldList) {
+    ObjectNode jsonNode = (ObjectNode) Jsons.jsonNode(Collections.emptyMap());
+    fieldList.forEach(field -> setJsonField(field, rowValues.get(field.getName()), jsonNode));
+    return jsonNode;
+  }
+
+  private static void setJsonField(Field field, FieldValue fieldValue, ObjectNode node) {
+    if (fieldValue.getAttribute().equals(Attribute.PRIMITIVE)) {
+      LegacySQLTypeName fieldType = field.getType();
+      String fieldName = field.getName();
+      if (fieldValue.isNull()) {
+        node.put(fieldName, (String) null);
+      } else
+        switch (fieldType.getStandardType()) {
+          case BOOL -> node.put(fieldName, fieldValue.getBooleanValue());
+          case INT64 -> node.put(fieldName, fieldValue.getLongValue());
+          case FLOAT64 -> node.put(fieldName, fieldValue.getDoubleValue());
+          case NUMERIC -> node.put(fieldName, fieldValue.getNumericValue());
+          case BIGNUMERIC -> node.put(fieldName, returnNullIfInvalid(fieldValue::getNumericValue));
+          case STRING -> node.put(fieldName, fieldValue.getStringValue());
+          case BYTES -> node.put(fieldName, fieldValue.getBytesValue());
+          case DATE -> node.put(fieldName, toISO8601String(getDateValue(fieldValue, BIG_QUERY_DATE_FORMAT)));
+          case DATETIME -> node.put(fieldName, toISO8601String(getDateValue(fieldValue, BIG_QUERY_DATETIME_FORMAT)));
+          case TIMESTAMP -> node.put(fieldName, toISO8601String(fieldValue.getTimestampValue() / 1000));
+          case TIME -> node.put(fieldName, fieldValue.getStringValue());
```

Comment:
why not make `struct` into an object and `array` into json schema `array` as well?

Reply:
done